notes/bytecode.md

```diff
@@ -0,0 +1,79 @@
+This note discusses ideas for initial NaNVM’s bytecode specification and bytecode interpreter design.
+
+### Basics
+
+We aim for a simple initial implementation that is parametrized by a set of options (that we name
+interpreter’s parameters now on). It’s interesting to run a set of benchmarks varying these parameters
+– to get insights on promising areas of improvements (to increase speed and then to reduce memory
+footprint – without sacrificing speed).
+
+“Bytecode” instructions are placed in one continuous array – not necessarily an array of bytes.
+We might get performance benefits from using a 2-byte or 4-byte or 8-byte instruction word.
```

Comment:
Some commands may contain pointers. A pointer (on 64-bit platform) has about 45-48 significant bits. If we want a fixed-size command, we should target at least 64 bits (8 bytes) per command.

Reply:
Makes sense, I will add this consideration to this initial bytecode.md commit, stating that we start with 8-bytes instructions. BTW here goes an idea on how such long instruction word can be used in an instruction that does not contain a pointer. A native call instruction might have several vacant bytes in its 8-byte instruction word. We can use that space for placing parameter values and / or references (in terms of 0-based index off the top of the data stack) in the instruction word, saving on copying parameters to the top of the stack. This trick does not work with bytecode calls.